### PR TITLE
JAMES-1975 Add DKIM integration test

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/configuration/MockPublicKeyRecordRetrieverModule.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package org.apache.james.mailets.configuration;
```
Comment (chibenwa) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/configuration/MockPublicKeyRecordRetrieverModule.java]: Missing license

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/DKIMIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            .sendMessage(FROM, RECIPIENT);
+
+        imapMessageReader.connect(LOCALHOST_IP, jamesServer.getProbe(ImapGuiceProbe.class).getImapPort())
+            .login(RECIPIENT, PASSWORD)
+            .select(IMAPMessageReader.INBOX)
+            .awaitMessage(awaitAtMostOneMinute);
+
+        assertThat(dkimAuthResults).hasSize(1);
+        assertThat(dkimAuthResults.get(0)).hasValueSatisfying(result -> assertThat(result).startsWith("pass"));
+        assertThat(imapMessageReader.readFirstMessage()).isNotNull();
```
Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/DKIMIntegrationTest.java]: why don't we build assertions on the mail we find with IMAP?

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/configuration/MockPublicKeyRecordRetrieverModule.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import org.apache.james.jdkim.api.PublicKeyRecordRetriever;
+
+import com.google.inject.AbstractModule;
+
+public class MockPublicKeyRecordRetrieverModule extends AbstractModule {
+    @Override
+    public void configure() {
+        super.configure();
+        bind(PublicKeyRecordRetriever.class).toInstance(new MockPublicKeyRecordRetriever(
+            "v=DKIM1; k=rsa; p=MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDYDaYKXzwVYwqWbLhmuJ66aTAN8wmDR+rfHE8HfnkSOax0oIoTM5zquZrTLo30870YMfYzxwfB6j/Nz3QdwrUD/t0YMYJiUKyWJnCKfZXHJBJ+yfRHr7oW+UW3cVo9CG2bBfIxsInwYe175g9UjyntJpWueqdEIo1c2bhv9Mp66QIDAQAB;",
```
Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/configuration/MockPublicKeyRecordRetrieverModule.java]: what is this value? Is it supposed to be a useful values for every cases?

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/DKIMIntegrationTest.java (hunk 2 above)
Comment (blackheaven) [reply server/mailet/integration-testing/src/test/java/org/apache/james/mailets/DKIMIntegrationTest.java]: That's a WIP, I was supposed to get the right value

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/configuration/PublicKeyRecordRetrieverModule.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package org.apache.james.mailets.configuration;
+
+import org.apache.james.jdkim.api.PublicKeyRecordRetriever;
+
+import com.google.inject.AbstractModule;
+
+public class PublicKeyRecordRetrieverModule extends AbstractModule {
```
Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/configuration/PublicKeyRecordRetrieverModule.java]: this whole class can be written as a lambda:
```java
binder -> binder.bind(PublicKeyRecordRetriever.class).toInstance(retriever);
```

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/DKIMIntegrationTest.java (hunk 2 above)
Comment (mbaechler) [reply server/mailet/integration-testing/src/test/java/org/apache/james/mailets/DKIMIntegrationTest.java]: My question was: why use a ExtractAttributeStub at all when you can read the headers in the mail you retrieve with IMAP?

**Comment by blackheaven** [comment]
> My question was: why use a ExtractAttributeStub at all when you can read the headers in the mail you retrieve with IMAP?

Because it's an attribute, not a header.

**Comment by blackheaven** [comment]
```
[INFO] Apache James :: Server :: Cassandra - guice injection FAILURE [02:54 min]
```

test this please

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/configuration/PublicKeyRecordRetrieverModule.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package org.apache.james.mailets.configuration;
```
Comment (chibenwa) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/configuration/PublicKeyRecordRetrieverModule.java]: Missing license?

**Comment by blackheaven** [comment]
> You probably lack the real implementation injection of PublicKeyRecordRetriever for cases where we really want to use that mailet

@mbaechler do we have a central place where to put it?

**Comment by blackheaven** [comment]
test this please

File: server/mailet/mailets-guice/pom.xml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>com.google.inject</groupId>
+            <artifactId>guice</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.james.jdkim</groupId>
+            <artifactId>apache-jdkim-library</artifactId>
+            <version>0.2</version>
+            <scope>compile</scope>
```
Comment (mbaechler) [inline server/mailet/mailets-guice/pom.xml]: compile is the default scope, isn't it?

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/DKIMIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+        dataProbe.addUser(RECIPIENT, PASSWORD);
+    }
+
+    @After
+    public void tearDown() {
+        jamesServer.shutdown();
+    }
+
+    @Test
+    public void incomingMessageShouldBeReceivedSigned() throws Exception {
```
Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/DKIMIntegrationTest.java]: the test name should explain that you want the DKIM verifier to check the signature.

File: server/container/guice/cassandra-guice/pom.xml [hunk 8]
```diff
@@ -81,6 +81,11 @@
             <groupId>${james.groupId}</groupId>
             <artifactId>james-server-mailbox-plugin-deleted-messages-vault-guice</artifactId>
         </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-mailets-guice</artifactId>
+            <scope>compile</scope>
```
Comment (chibenwa) [inline server/container/guice/cassandra-guice/pom.xml]: Why is scope compile needed for?

File: server/mailet/mailets-guice/src/main/java/org/apache/james/mailets/configuration/MailetsConfigurationModule.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+import org.apache.james.jdkim.api.PublicKeyRecordRetriever;
+import org.apache.james.jdkim.impl.DNSPublicKeyRecordRetriever;
+import org.apache.james.jdkim.impl.MultiplexingPublicKeyRecordRetriever;
+
+import com.google.inject.AbstractModule;
+
+public class MailetsConfigurationModule extends AbstractModule {
+    @Override
+    public void configure() {
+        super.configure();
```
Comment (chibenwa) [inline server/mailet/mailets-guice/src/main/java/org/apache/james/mailets/configuration/MailetsConfigurationModule.java]: Remove this useless call to an empty method please

**Comment by trantienduchn** [comment]
```
[014d46f9cd1d9f8cb5a1e22d888b411be29a7c36] [ERROR] Errors: 
[014d46f9cd1d9f8cb5a1e22d888b411be29a7c36] [ERROR]   RabbitMQEventBusTest$LifeCycleTest$SingleEventBus.dispatchShouldWorkAfterRestartForOldRegistration:357 » ConditionTimeout
[014d46f9cd1d9f8cb5a1e22d888b411be29a7c36] [ERROR]   RabbitMQEventBusTest$LifeCycleTest$SingleEventBus.registrationsShouldNotHandleEventsAfterStop » ThrownByLambda
```
test this please

**Comment by blackheaven** [comment]
```
[ERROR]   RabbitMQEventBusTest$LifeCycleTest$SingleEventBus$DispatchingWhenNetWorkIssue.dispatchShouldWorkAfterNetworkIssuesForOldRegistration:333 » ConditionTimeout
```

test this please

**Comment by blackheaven** [comment]
test this please

**Comment by blackheaven** [comment]
test this please

**Comment by mbaechler** [comment]
test this please

**Comment by chibenwa** [comment]
```

[014d46f9cd1d9f8cb5a1e22d888b411be29a7c36] [ERROR] Failures: 
[014d46f9cd1d9f8cb5a1e22d888b411be29a7c36] [ERROR]   AbstractSieveRepositoryTest.setActiveShouldWork:152 
[014d46f9cd1d9f8cb5a1e22d888b411be29a7c36] Expecting:
[014d46f9cd1d9f8cb5a1e22d888b411be29a7c36]   <[ScriptSummary{name=ScriptName{value=script}, activeFile=false}]>
[014d46f9cd1d9f8cb5a1e22d888b411be29a7c36] to contain only:
[014d46f9cd1d9f8cb5a1e22d888b411be29a7c36]   <[ScriptSummary{name=ScriptName{value=script}, activeFile=true}]>
[014d46f9cd1d9f8cb5a1e22d888b411be29a7c36] elements not found:
[014d46f9cd1d9f8cb5a1e22d888b411be29a7c36]   <[ScriptSummary{name=ScriptName{value=script}, activeFile=true}]>
[014d46f9cd1d9f8cb5a1e22d888b411be29a7c36] and elements not expected:
[014d46f9cd1d9f8cb5a1e22d888b411be29a7c36]   <[ScriptSummary{name=ScriptName{value=script}, activeFile=false}]>
```

**Comment by mbaechler** [comment]
test this please

**Comment by blackheaven** [comment]
```
[INFO] Apache James :: Mailbox :: Zookeeper Sequence Provider FAILURE [  1.850 s]
```

test this please

**Comment by chibenwa** [comment]
`Zookeeper Sequence Provider` looks deprecable?

**Comment by chibenwa** [comment]
`Zookeeper Sequence Provider` looks deprecable?

`CassandraMailRepositoryIntegrationTest.deniedSenderMailShouldBeStoredInCassandraMailRepositoryWhenConfigured:72 » SMTPSending`

test this please

**Comment by blackheaven** [comment]
```
[ERROR]   CassandraMailRepositoryIntegrationTest.deniedSenderMailShouldBeStoredInCassandraMailRepositoryWhenConfigured:72 » SMTPSending
```

test this please

**Comment by blackheaven** [comment]
cassandra

test this please

**Comment by chibenwa** [comment]
`MailRepositoryContract.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult:474 » Execution`

test this please

**Comment by mbaechler** [comment]
```
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] java.lang.NoSuchMethodError: org.apache.mailet.Mail.setAttribute(Lorg/apache/mailet/Attribute;)Ljava/util/Optional;
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.apache.james.smtpserver.AddDefaultAttributesMessageHook.onMessage(AddDefaultAttributesMessageHook.java:62)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.apache.james.smtpserver.DataLineJamesMessageHookHandler.processExtensions(DataLineJamesMessageHookHandler.java:184)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.apache.james.smtpserver.DataLineJamesMessageHookHandler.onLine(DataLineJamesMessageHookHandler.java:118)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.apache.james.protocols.smtp.core.DataCmdHandler$DataLineFilterWrapper.onLine(DataCmdHandler.java:101)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.apache.james.protocols.smtp.core.DataCmdHandler$DataLineFilterWrapper.onLine(DataCmdHandler.java:88)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.apache.james.protocols.smtp.core.SeparatingDataLineFilter.onBodyLine(SeparatingDataLineFilter.java:108)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.apache.james.protocols.smtp.core.SeparatingDataLineFilter.onLine(SeparatingDataLineFilter.java:66)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.apache.james.protocols.smtp.core.DataCmdHandler$DataLineFilterWrapper.onLine(DataCmdHandler.java:101)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.apache.james.protocols.smtp.core.DataCmdHandler$DataLineFilterWrapper.onLine(DataCmdHandler.java:88)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.apache.james.protocols.smtp.core.esmtp.MailSizeEsmtpExtension.onLine(MailSizeEsmtpExtension.java:164)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.apache.james.protocols.smtp.core.DataCmdHandler$DataLineFilterWrapper.onLine(DataCmdHandler.java:101)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.apache.james.protocols.smtp.core.DataCmdHandler$DataLineFilterWrapper.onLine(DataCmdHandler.java:88)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.apache.james.protocols.netty.LineHandlerUpstreamHandler.messageReceived(LineHandlerUpstreamHandler.java:50)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.messageReceived(SimpleChannelUpstreamHandler.java:124)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.jboss.netty.handler.execution.ChannelUpstreamEventRunnable.doRun(ChannelUpstreamEventRunnable.java:43)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.jboss.netty.handler.execution.ChannelEventRunnable.run(ChannelEventRunnable.java:67)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.jboss.netty.handler.execution.OrderedMemoryAwareThreadPoolExecutor$ChildExecutor.run(OrderedMemoryAwareThreadPoolExecutor.java:314)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at java.lang.Thread.run(Thread.java:748)
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.449 s <<< FAILURE! - in org.apache.james.CassandraMailRepositoryIntegrationTest
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] [ERROR] deniedSenderMailShouldBeStoredInCassandraMailRepositoryWhenConfigured{GuiceJamesServer}  Time elapsed: 0.442 s  <<< ERROR!
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] org.apache.james.utils.SMTPSendingException: 
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] Error upon step Data: 451 Unable to process request
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 
[6d9d186ff2c8b111001f61ef18dde11e2d63060a] 	at org.apache.james.CassandraMailRepositoryIntegrationTest.deniedSenderMailShouldBeStoredInCassandraMailRepositoryWhenConfigured(CassandraMailRepositoryIntegrationTest.java:72)
```

This looks like a real problem to fix: dependency problem we had in previous DKIM PR.

File: server/mailet/integration-testing/pom.xml [hunk 10]
```diff
@@ -147,6 +151,38 @@
             <artifactId>json-unit-assertj</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (mbaechler) [inline server/mailet/integration-testing/pom.xml]: why do we need these libraries instead of dkim mailet?

File: server/pom.xml [hunk 11]
```diff
@@ -77,6 +77,7 @@
         <module>mailet/mailetcontainer-api</module>
         <module>mailet/mailetcontainer-camel</module>
         <module>mailet/mailets</module>
+        <module>mailet/mailets-guice</module>
```
Comment (mbaechler) [inline server/pom.xml]: guice should be located only in server/container/guice

**Comment by blackheaven** [comment]
```
[INFO] Apache James :: Server :: JMAP :: Integration testing Common SKIPPED
```

why?

test this please

**Comment by blackheaven** [comment]
```
[ERROR]   RabbitMQEventBusTest$LifeCycleTest$SingleEventBus$DispatchingWhenNetWorkIssue.dispatchShouldWorkAfterNetworkIssuesForOldRegistration:366 » ConditionTimeout
```

test this please

File: server/container/guice/mailet/pom.xml [hunk 12]
```diff
@@ -94,6 +94,11 @@
             <artifactId>assertj-core</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.james.jdkim</groupId>
+            <artifactId>apache-jdkim-library</artifactId>
```
Comment (mbaechler) [inline server/container/guice/mailet/pom.xml]: Why do we need that dependency?

Comment (blackheaven) [reply server/container/guice/mailet/pom.xml]: `PublicKeyRecordRetriever`, `MultiplexingPublicKeyRecordRetriever` and `DNSPublicKeyRecordRetriever`

**Comment by blackheaven** [comment]
test this please

File: server/container/guice/mailet/pom.xml (hunk 12 above)
Comment (mbaechler) [reply server/container/guice/mailet/pom.xml]: Well, doesn't dkim mailet have a transitive dependency?

**Comment by blackheaven** [comment]
Still got the error:

```
java.lang.NoSuchMethodError: org.apache.mailet.Mail.setAttribute(Lorg/apache/mailet/Attribute;)Ljava/util/Optional;
```

@mbaechler what was your fix?

**Comment by mbaechler** [comment]
> @mbaechler what was your fix?

apache-jdkim-library imports conflicting version of mailet-api, you have to exclude some libraries.
However, just fixing https://github.com/linagora/james-project/pull/2383#discussion_r296631472 should fix your issue.

**Comment by mbaechler** [comment]
I pushed the fix

**Comment by chibenwa** [comment]
```
[252330f28294f63db095ddaf0495a1f4cf889525] [ERROR] noAckShouldNotDecreaseSize  Time elapsed: 0.227 s  <<< ERROR!
[252330f28294f63db095ddaf0495a1f4cf889525] reactor.rabbitmq.RabbitFluxException: Not retryable exception, cannot retry
[252330f28294f63db095ddaf0495a1f4cf889525] 	at org.apache.james.queue.api.ManageableMailQueueContract.noAckShouldNotDecreaseSize(ManageableMailQueueContract.java:93)
[252330f28294f63db095ddaf0495a1f4cf889525] Caused by: com.rabbitmq.client.AlreadyClosedException: channel is already closed due to clean channel shutdown; protocol method: #method<channel.close>(reply-code=200, reply-text=OK, class-id=0, method-id=0)
[252330f28294f63db095ddaf0495a1f4cf889525] 	at org.apache.james.queue.api.ManageableMailQueueContract.noAckShouldNotDecreaseSize(ManageableMailQueueContract.java:93)
[252330f28294f63db095ddaf0495a1f4cf889525] 
```

test this please

**Comment by Arsnael** [comment]
```
[252330f28294f63db095ddaf0495a1f4cf889525] [ERROR] Errors: 
[252330f28294f63db095ddaf0495a1f4cf889525] [ERROR]   ManageableMailQueueContract.noAckShouldNotDecreaseSize:93 » RabbitFlux Not ret...
```
test this please

**Comment by blackheaven** [comment]
```

[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR] Errors: 
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   LinshareBlobExportMechanismIntegrationTest.exportShouldSendAnEmailToShareeWhenImapDelete » NoClassDefFound
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   LinshareBlobExportMechanismIntegrationTest.exportShouldSendAnEmailToShareeWhenJmapDelete » NoClassDefFound
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   LinshareBlobExportMechanismIntegrationTest.exportShouldShareNonEmptyZipViaLinshareWhenImapDelete » NoClassDefFound
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   LinshareBlobExportMechanismIntegrationTest.exportShouldShareNonEmptyZipViaLinshareWhenJmapDelete » NoClassDefFound
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   LinshareBlobExportMechanismIntegrationTest.exportShouldShareTheDocumentViaLinshareWhenImapDelete » NoClassDefFound
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   LinshareBlobExportMechanismIntegrationTest.<clinit>:85 » ExceptionInInitializer
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   SpamAssassinContract.deletingSpamMessageShouldNotImpactSpamAssassinState:557 » ConditionTimeout
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   SpamAssassinContract.expungingSpamMessageShouldNotImpactSpamAssassinState:493 » ConditionTimeout
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   SpamAssassinContract.imapCopiesToSpamMailboxShouldBeConsideredAsSpam:194 » ConditionTimeout
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   SpamAssassinContract.imapMovesToSpamMailboxShouldBeConsideredAsSpam:242 » ConditionTimeout
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   SpamAssassinContract.spamShouldBeDeliveredInSpamMailboxOrInboxWhenMultipleRecipientsConfigurations:628 » ConditionTimeout
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   SpamAssassinContract.spamShouldBeDeliveredInSpamMailboxWhenSameMessageHasAlreadyBeenMovedToSpam:146 » ConditionTimeout
```

test this please

**Comment by blackheaven** [comment]
```
java.io.IOException: Server returned HTTP response code: 500 for URL: https://api.github.com/repos/linagora/james-project/statuses/d572a5de954a6c9d6e2bf46705de4713a0cff64f
```

test this please

**Comment by Arsnael** [comment]
```
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR] Errors: 
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   LinshareBlobExportMechanismIntegrationTest.exportShouldSendAnEmailToShareeWhenImapDelete » NoClassDefFound
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   LinshareBlobExportMechanismIntegrationTest.exportShouldSendAnEmailToShareeWhenJmapDelete » NoClassDefFound
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   LinshareBlobExportMechanismIntegrationTest.exportShouldShareNonEmptyZipViaLinshareWhenImapDelete » NoClassDefFound
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   LinshareBlobExportMechanismIntegrationTest.exportShouldShareNonEmptyZipViaLinshareWhenJmapDelete » NoClassDefFound
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   LinshareBlobExportMechanismIntegrationTest.exportShouldShareTheDocumentViaLinshareWhenImapDelete » NoClassDefFound
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   LinshareBlobExportMechanismIntegrationTest.<clinit>:85 » ExceptionInInitializer
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   SpamAssassinContract.deletingSpamMessageShouldNotImpactSpamAssassinState:557 » ConditionTimeout
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   SpamAssassinContract.expungingSpamMessageShouldNotImpactSpamAssassinState:493 » ConditionTimeout
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   SpamAssassinContract.imapCopiesToSpamMailboxShouldBeConsideredAsSpam:194 » ConditionTimeout
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   SpamAssassinContract.imapMovesToSpamMailboxShouldBeConsideredAsSpam:242 » ConditionTimeout
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   SpamAssassinContract.movingAMailToTrashShouldNotImpactSpamassassinLearning:370 » ConditionTimeout
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   SpamAssassinContract.spamShouldBeDeliveredInSpamMailboxOrInboxWhenMultipleRecipientsConfigurations:628 » ConditionTimeout
[d572a5de954a6c9d6e2bf46705de4713a0cff64f] [ERROR]   SpamAssassinContract.spamShouldBeDeliveredInSpamMailboxWhenSameMessageHasAlreadyBeenMovedToSpam:146 » ConditionTimeout
```
test this please

**Pull request merged**